### PR TITLE
AUTH-244 encode uri for find options

### PR DESCRIPTION
**Pull request by ahmadfaridulazmi**
apply uri encoded to `find` options

### CODE REVIEW

File: spec/volcanic/authenticator/v1/identity_spec.rb [hunk 1]
```diff
@@ -205,5 +205,15 @@
...
+
+    context 'when find with name' do
+      subject(:identities) { identity.find(name: 'volcanic') }
+      it { expect(identities.first.name).to eq 'volcanic' }
+    end
+
+    context 'when find with name that has special character' do
+      subject(:identities) { identity.find(name: 'volcanic_ý') }
+      it { expect(identities.first.name).to eq 'volcanic_ý' }
+    end
```
Comment (danleyden) [inline spec/volcanic/authenticator/v1/identity_spec.rb]: another test please to prove what happens when sending with a '+' character in the name, e.g. `volcanic+1234`

File: spec/volcanic/authenticator/v1/principal_spec.rb [hunk 2]
```diff
@@ -67,6 +67,16 @@
...
+
+    context 'when find with name' do
+      subject(:principals) { principal.find(name: 'volcanic') }
+      it { expect(principals.first.name).to eq 'volcanic' }
+    end
+
+    context 'when find with name that has special character' do
+      subject(:principals) { principal.find(name: 'volcanic_ý') }
+      it { expect(principals.first.name).to eq 'volcanic_ý' }
+    end
```
Comment (danleyden) [inline spec/volcanic/authenticator/v1/principal_spec.rb]: same as with identity - another test with '+' character please

File: spec/volcanic/authenticator/v1/identity_spec.rb (hunk 1 above)
Comment (ahmadfaridulazmi) [reply spec/volcanic/authenticator/v1/identity_spec.rb]: @danleyden  done

File: spec/volcanic/authenticator/v1/principal_spec.rb (hunk 2 above)
Comment (ahmadfaridulazmi) [reply spec/volcanic/authenticator/v1/principal_spec.rb]: same here

**Comment by danleyden** [comment]
@ahmadfaridulazmi can you squash and rebase this change before merge please?

**Comment by danleyden** [comment]
not sure exactly what got changed.... but I was expecting to end up with 1-2 commits (only) as part of the PR rather than all of them

**Pull request merged**